arches_lingo/src/arches_lingo/components/scheme/report/SchemeStandard.vue

```diff
@@ -1,20 +1,122 @@
 import SchemeReportSection from "@/arches_lingo/components/scheme/report/SchemeSection.vue";
+import {
+    fetchSchemeCreation,
+    fetchTextualWorkRdmSystemList,
+    updateSchemeCreation,
+} from "@/arches_lingo/api.ts";
+import ResourceInstanceRelationships from "../../generic/ResourceInstanceRelationships.vue";
+import { selectedLanguageKey, VIEW, EDIT } from "@/arches_lingo/constants.ts";
+import type { Language } from "@/arches_vue_utils/types.ts";
 
-const save = () => {
-    console.log("save");
-};
-const getSectionValue = async () => {
-    console.log("update");
-};
+const OPEN_EDITOR = "openEditor";
+const schemeInstance = ref<SchemeInstance>();
+const textualWorkOptions = ref<ResourceInstanceReference[]>();
+const route = useRoute();
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+const { $gettext } = useGettext();
+
+defineProps<{
+    mode?: DataComponentMode;
+}>();
+
+defineEmits(["openEditor"]);
 
 defineExpose({ save, getSectionValue });
-const { $gettext } = useGettext();
+
+onMounted(async () => {
+    getSectionValue();
+});
+
+async function getOptions(): Promise<ResourceInstanceReference[]> {
+    const options = await fetchTextualWorkRdmSystemList();
+    const results = options.map((option: ResourceInstanceResult) => {
+        const result: ResourceInstanceReference = {
+            display_value: option.descriptors[selectedLanguage.value.code].name,
+            resourceId: option.resourceinstanceid,
+            ontologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
+            inverseOntologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
+        };
+        return result;
+    });
+    return results;
+}
+
+async function save() {
+    await updateSchemeCreation(
+        route.params.id as string,
+        schemeInstance.value as SchemeInstance,
+    );
+
+    getSectionValue();
+}
+
+async function getSectionValue() {
+    const options = !textualWorkOptions.value
+        ? await getOptions()
+        : textualWorkOptions.value;
+
+    const scheme = await fetchSchemeCreation(route.params.id as string);
+
+    const hydratedResults = options.map((option) => {
+        const savedSource = scheme.creation.creation_sources_.find(
+            (source: ResourceInstanceReference) =>
+                source.resourceId === option.resourceId,
+        );
+        if (savedSource) {
+            return savedSource;
+        } else {
+            return option;
+        }
+    });
+    textualWorkOptions.value = hydratedResults;
+    schemeInstance.value = scheme;
+}
+
+function onCreationUpdate(val: string[]) {
+    const schemeInstanceValue = schemeInstance.value as SchemeInstance;
+    if (textualWorkOptions.value) {
+        const options = textualWorkOptions.value?.filter((option) =>
+            val.includes(option.resourceId),
+        );
+        if (!schemeInstanceValue?.creation) {
+            schemeInstanceValue.creation = { creation_sources_: options };
```

Comment:
Heads up that the trailing underscores are removed in main now.

Reply:
nice

Reply:
updated